windows-gaming-driver/src/driver/control/codec.rs

```diff
@@ -50,3 +52,40 @@ impl Encoder for Codec {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use bytes::BytesMut;
+    use tokio_io::codec::Decoder;
+
+    macro_rules! test {
+        (fn $name:ident, with $data:expr, expect $expected:expr, len $len:expr) => (
```

Comment:
Is it, I wonder? I mean this macro is a straight up function / string replace. Literally the only reason this is a macro instead of just a function is to have less boilerplate code as the macro can generate the `fn $name() {}` for you. There is nothing complex about this, nothing one could possibly misunderstand. What remains is a normal function call and Rust doesn't have named parameters. What I'm trying to say here is that this is equivalent to defining a convention like this:

```rust
let mut bytes = BytesMut::new();
bytes.extend(/* buffer: */ &[1,2]);
assert_eq!(/* left: */ Codec.decode(/* buffer: */ &mut bytes).unwrap(), /* right: */ Some(/* value: */ ControlCmdIn::IoEntry));
assert_eq!(/* left: */ bytes.len(), /* right: */ 1);
assert_eq!(/* left: */ Codec.decode(/* buffer: */ &mut bytes).unwrap(), /* right: */ Some(/* value: */ ControlCmdIn::Shutdown));
assert_eq!(/* left: */ bytes.len(), /* right: */ 0);
```

And this is **clearly** overly verbose.